### PR TITLE
feat: ParticleRenderer add isAutoplay and blendMode property

### PR DESCRIPTION
**Pull request by eyworldwide**
### Please check if the PR fulfills these requirements

- [X] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
- Feature: ParticleRenderer add `isAutoplay` and `blendMode` property.
- Bug fix: texture uv Y flip bug. 

### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
No.
### Other information:

### CODE REVIEW

**Review by jserranoq — APPROVED** [review]

**Comment by jserranoq** [comment]
Mm

**Review by jserranoq — APPROVED** [review]

**Review by jserranoq — APPROVED** [review]

**Review by jserranoq — APPROVED** [review]

**Review by jserranoq — APPROVED** [review]

**Review by jserranoq — APPROVED** [review]

**Comment by jserranoq** [comment]
> ### Please check if the PR fulfills these requirements
> 
> 
> 
> - [X] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
> 
> - [ ] Tests for the changes have been added (for bug fixes / features)
> 
> - [ ] Docs have been added / updated (for bug fixes / features)
> 
> 
> 
> ### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
> 
> - Feature: ParticleRenderer add `isAutoplay` and `blendMode` property.
> 
> - Bug fix: texture uv Y flip bug. 
> 
> 
> 
> ### What is the current behavior? (You can also link to an open issue here)
> 
> 
> 
> ### What is the new behavior (if this is a feature change)?
> 
> 
> 
> ### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
> 
> No.
> 
> ### Other information:

**Review by singlecoder — APPROVED** [review]
+1

File: packages/core/src/particle/ParticleRenderer.ts [hunk 1]
```diff
@@ -27,6 +27,11 @@ enum DirtyFlagType {
   Everything = 0xffffffff
 }
 
+export enum ParticleRendererBlendMode {
```
Comment (GuoLei1990) [inline packages/core/src/particle/ParticleRenderer.ts:30]: Write comments.

File: packages/core/src/particle/ParticleRenderer.ts [hunk 2]
```diff
@@ -464,6 +478,47 @@ export class ParticleRenderer extends GeometryRenderer {
...
+    }
+    else if (value === ParticleRendererBlendMode.Additive) {
+      target.sourceColorBlendFactor = target.sourceAlphaBlendFactor = BlendFactor.SourceAlpha;
+      target.destinationColorBlendFactor = target.destinationAlphaBlendFactor = BlendFactor.One;
+    }
+
+    this._blendMode = value;
+  }
+
   constructor(props) {
```
Comment (GuoLei1990) [inline packages/core/src/particle/ParticleRenderer.ts:522]: add @internal comments

File: packages/core/src/particle/ParticleRenderer.ts [hunk 3]
```diff
@@ -464,6 +478,47 @@ export class ParticleRenderer extends GeometryRenderer {
     this._isFadeOut = value;
   }
 
+  /**
+   * Is auto play or not
```
Comment (GuoLei1990) [inline packages/core/src/particle/ParticleRenderer.ts:482]: write .

File: packages/core/src/particle/ParticleRenderer.ts [hunk 4]
```diff
@@ -343,11 +350,16 @@ export class ParticleRenderer extends GeometryRenderer {
...
     this._isInit = false;
     this._maxCount = value;
+    this._updateDirtyFlag = DirtyFlagType.Everything;
     this.geometry = this._createGeometry();
 
     this._updateBuffer();
 
     this._isInit = true;
+
+    if (this._isAutoplay) {
```
Comment (GuoLei1990) [inline packages/core/src/particle/ParticleRenderer.ts:360]: isAutoplay should call play on onEnable event,should not call on set maxCount. 
The name can more detail,for exmaple "playOnEnabled".

**Comment by eyworldwide** [comment]
This PR trigger some bug of github. New PR here：https://github.com/oasis-engine/engine/pull/56

**Pull request closed without merging**